frontend/src/metabase/querying/notebook/components/NotebookDataPicker/EmbeddingDataPicker/EmbeddingDataPicker.tsx

```diff
@@ -74,14 +89,22 @@ export function EmbeddingDataPicker({
     );
   }
 
+  const isSourceSelected = Boolean(pickerInfo?.tableId);
   return (
     <PLUGIN_EMBEDDING.DataSourceSelector
-      key={pickerInfo?.tableId}
-      isInitiallyOpen={!table}
+      key={
+        isSourceSelected
+          ? pickerInfo?.tableId
+          : `${sourceTable?.id}:${isSourceModelFetching}`
```

Comment:
when we're in the join step, initially, the `pickerInfo` would not have value, until we actually select any data source in the join step (when the join step still says `Pick data...`).

![image](https://github.com/user-attachments/assets/4a4fef0b-1769-4d55-91ac-992fbc28a912)

When we're in this state, we want to make sure when changing the data source ID. I also make `isSourceModelFetching` into the key to force the data picker to remount when we get the new model card since that's when we'll know the source model collection ID.

`DataSourceSelector` isn't really good at handling prop changing without remounting, so this `key` shenanigan is needed.